### PR TITLE
feat(docs-hyper-link): hyper link refactor

### PR DESCRIPTION
**Pull request by weird94**
close https://github.com/dream-num/univer-pro/issues/2156

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Comment by oumomomo** [comment]
对.设置超链接，展示tips时无法点击按钮
image

**Comment by oumomomo** [comment]
页眉内输入内容设置超链接勾选首页不同，点击第二页的编辑不展示面板
[2024_8_22 17_04_24 video.webm](https://github.com/user-attachments/assets/d76d9a70-dae3-4784-950d-41b3485315fd)

File: packages/core/src/services/command/command.service.ts [hunk 1]
```diff
@@ -411,7 +411,8 @@ export class CommandService extends Disposable implements ICommandService {
     private async _execute<P extends object, R = boolean>(command: ICommand<P, R>, params?: P, options?: IExecutionOptions): Promise<R> {
         this._logService.debug(
             '[CommandService]',
-            `${'|-'.repeat(Math.max(this._commandExecutingLevel, 0))}executing command "${command.id}"`
+            `${'|-'.repeat(Math.max(this._commandExecutingLevel, 0))}executing command "${command.id}"`,
+            { params }
```
Comment (wzhudev) [inline packages/core/src/services/command/command.service.ts:415]: Do not expose params.

File: packages/core/src/types/interfaces/i-document-data.ts [hunk 2]
```diff
@@ -338,23 +338,23 @@ export interface ITextRun {
...
     startIndex: number;
     endIndex: number;
     rangeId: string;
-    rangeType: CustomRangeType;
+    rangeType: CustomRangeType | number;
     /**
      * display as a whole-entity
      */
     wholeEntity?: boolean;
+    properties?: T;
```
Comment (wzhudev) [inline packages/core/src/types/interfaces/i-document-data.ts:353]: JSDoc required.

**Review by wzhudev — CHANGES_REQUESTED** [review]
This PR seems introducing a huge break change into the doc system. I would like to know the following before continue reviewing the code.

* What's the plan to deal with the legacy resources in old snapshots? Will they are lost?
* What users have to change in their code? Are there any breaking changes not documented in the commit messages?

**Comment by weird94** [comment]
> This PR seems introducing a huge break change into the doc system. I would like to know the following before continue reviewing the code.
> 
> * What's the plan to deal with the legacy resources in old snapshots? Will they are lost?
> * What users have to change in their code? Are there any breaking changes not documented in the commit messages?
> 
> And this PR's title seems no accurate as not only the hyper link feature was refactored.

1. this change is compatible with the old snapshot, `DocHyperLinkResourceController` will load links from snapshot and sync to custom-ranges.

**Review by wzhudev — APPROVED** [review]
LGTM

**Pull request merged**